File: leetcode/448.找到所有数组中消失的数字/448.找到所有数组中消失的数字.go
package main

import "fmt"

func findDisappearedNumbers(nums []int) []int {
	n := len(nums)
	res := make([]int, 0, n)
	numsMap := make(map[int]int8, n)
	for _, num := range nums {
		numsMap[num] = 0
	}
	for i := 1; i < n+1; i++ {
		if _, ok := numsMap[i]; !ok {
			res = append(res, i)
		}
	}
	return res
}

func main() {
	nums := []int{4, 3, 2, 7, 8, 2, 3, 1}
	result := findDisappearedNumbers(nums)
	fmt.Println(result)
}

File: leetcode/448.找到所有数组中消失的数字/448.找到所有数组中消失的数字.py
from typing import List

"""
遍历用set进行保存
再生成1~n+1的数值，进行查找，如果不存在就存入结果集
"""


class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        res = []
        n = len(nums)
        nums_set = set()
        for num in nums:
            nums_set.add(num)

        for i in range(1, n + 1):
            if i not in nums_set:
                res.append(i)

        return res


if __name__ == '__main__':
    nums = [4, 3, 2, 7, 8, 2, 3, 1]
    result = Solution().findDisappearedNumbers(nums)
    print(result)
